packages/db-postgres/src/queries/getTableColumnFromPath.ts

```diff
@@ -317,21 +317,15 @@ export const getTableColumnFromPath = ({
 
         // Join in the relationships table
         joinAliases.push({
-          condition: eq(
-            (aliasTable || adapter.tables[rootTableName]).id,
-            aliasRelationshipTable.parent,
+          condition: and(
+            eq((aliasTable || adapter.tables[rootTableName]).id, aliasRelationshipTable.parent),
+            like(aliasRelationshipTable.path, `${constraintPath}${field.name}`),
           ),
           table: aliasRelationshipTable,
         })
 
         selectFields[`${relationTableName}.path`] = aliasRelationshipTable.path
 
-        constraints.push({
```

Comment:
Why are we removing the path here?
The purpose of the path constraint is to make sure that the field being queried is matched. Consider if you had two relationship fields on one document, there would be no way to discern without the path.

Reply:
It's moved into the join constraint above to avoid filtering out rows that doesn't have relations. If there would be no relations the joined colums will be null instead, allowing us to select rows with unset relation.

Reply:
Oh yes, I see that now. Thanks.